rules/windows/powershell/powershell_script/posh_ps_domainpasswordspray_attacks.yml

```diff
@@ -0,0 +1,33 @@
+title: DomainPasswordSpray Attacks
+id: 5ff65fa4-32a1-4773-90cc-6238bf784148
+status: test
+description: Detects performing a domain password spray using the DomainPasswordSpray tool.
+references:
+    - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1110.003/T1110.003.md
+    - https://github.com/dafthack/DomainPasswordSpray
+author: Mustafa Kaan Demir (mkdemir)
+tags:
+    - attack.execution
+    - attack.t1110.003
+    - attack.t1059.001
+logsource:
+    product: windows
+    category: ps_script
+    definition: Script Block Logging must be enabled
+detection:
+    selection:
+        ScriptBlockText|contains:
+           - 'Invoke-DomainPasswordSpray'
+    selection_p1:
+        ScriptBlockText|contains:
+        - ' -Password'
+    selection_p2:
+        ScriptBlockText|contains:
+        - ' -UserList'
+        - ' -PasswordList'
+        - ' -Domain'
+        - ' -Force'
+    condition: selection and (selection_p1 or selection_p2)
```

Comment:
* as it is (selection_p1 or selection_p2) merge it contains is a "or"
* List (use of "-") with only 1 element is not a good way 
* Use 4 spaces for indentation

```yaml
    selection_tools:
        ScriptBlockText|contains: 'Invoke-DomainPasswordSpray'
    selection_option:
        ScriptBlockText|contains:
            - ' -Password'
            - ' -UserList'
            - ' -PasswordList'
            - ' -Domain'
            - ' -Force'
    condition: all of selection_*
```

ScriptBlockText have the launch option in it ?

Reply:
I think since it's an `and` we could drop the rest of the arguments. Since the Function `Invoke-DomainPasswordSpray` is unique enough to identify malicious usage.

So
```yml
selection:
        ScriptBlockText|contains: 'Invoke-DomainPasswordSpray'
```